src/doc/rustdoc/src/linking-to-items-by-name.md

```diff
@@ -89,6 +114,32 @@ struct Foo {}
 fn Foo() {}
 ```
 
-Note: Because of how `macro_rules!` macros are scoped in Rust, the intra-doc links of a `macro_rules!` macro will be resolved [relative to the crate root][#72243], as opposed to the module it is defined in.
+## Warnings, re-exports, and scoping
+
+Links are resolved in the current module scope, even when re-exported. If a link from another
+crate fails to resolve, no warning is given.
+
+When re-exporting an item, rustdoc allows additional documentation to it. That documentation will
+be resolved in the new scope, not the original, allowing you to link to items in the current
+crate. The new links will still give a warning if they fail to resolve.
+
+```rust
+/// See also [foo()]
+pub use std::process::Command;
+
+pub fn foo() {}
+```
+
+This is especially useful for proc-macros, which must always be in their own dedicated crate.
+
+Note: Because of how `macro_rules!` macros are scoped in Rust, the intra-doc links of a
+`macro_rules!` macro will be resolved [relative to the crate root][#72243], as opposed to the
+module it is defined in.
+
+If links do not look 'sufficiently like' an intra-doc link, they will be ignored and no warning
+will be given, even if the link fails to resolve. For example, any link containing `/` or `[]`
+characters will be ignored. You can see the full criteria for 'sufficiently like' in [the source
+code].
```

Comment:
I'm not sure if it's a good idea to link to the source code here. For one, it will very easily become out-of-date because it's tied to a particular commit. I think we should either skip this or include the list directly in the documentation.

Reply:
> Maybe I could change the commit hash to `master` instead? That might have wrong line numbers though.

Yeah, I think that would be very confusing.

> Hmm, I kind of prefer not to tie this down so we can update it in the future.

What do you mean? Documenting it isn't "tying it down".

Reply:
So, this is a not a reference, this is a _book_, and I don't think it should go into that much detail. We do not need to document every last corner of the implementation here. Let's just drop this section.